Library/Homebrew/cli/error.rb

```diff
@@ -0,0 +1,73 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "utils/formatter"
+
+module Homebrew
+  module CLI
+    class OptionConstraintError < UsageError
```

Comment:
Mostly unrelated to the PR, these error classes are moved (unchanged) out of `cli/parser.rb` to reduce file size / cognitive load.

Reply:
@dduugg Might be worth splitting into a separate PR to get merged earlier and slim the diff here?

Reply:
Done!